src/common/AdaptiveTree.cpp

```diff
@@ -260,16 +260,16 @@ namespace adaptive
     // We assume that Stop() method has been called before the deconstruction
     m_threadStop = true;
 
-    if (m_thread.joinable())
-      m_thread.join();
+    if (m_future.valid())
+      m_future.wait();
   }
 
   void AdaptiveTree::TreeUpdateThread::Initialize(AdaptiveTree* tree)
   {
-    if (!m_thread.joinable())
+    if (!m_future.valid())
     {
       m_tree = tree;
-      m_thread = std::thread(&TreeUpdateThread::Worker, this);
+      m_future = GlobalThreadPool.Execute(&TreeUpdateThread::Worker, this);
```

Comment:
please avoid do these changes to the update thread, this is a single thread not multiple, so it start on playback starts and nothing more

another reason is that the update thread worker must be in the future redone from scratch because currently implementation is not suitable for handling all kind of manifest updates, so i prefer keep its complexity lower to make easier future reworks

Reply:
Indeed. No clue why I identified this a source of (many) spawning threads.